src/main/java/bio/terra/service/filedata/flight/ingest/IngestPopulateFileStateFromFileGcpStep.java

```diff
@@ -42,10 +46,16 @@ public StepResult doStep(FlightContext context) {
     FlightMap inputParameters = context.getInputParameters();
     BulkLoadRequestModel loadRequest =
         inputParameters.get(JobMapKeys.REQUEST.getKeyName(), BulkLoadRequestModel.class);
-    GoogleBucketResource bucketResource =
-        FlightUtils.getContextValue(context, FileMapKeys.BUCKET_INFO, GoogleBucketResource.class);
-    Storage storage = gcsPdao.storageForBucket(bucketResource);
-    String projectId = bucketResource.projectIdForBucket();
+
+    String projectId;
+    if (dataset.isSelfHosted()) {
+      projectId = dataset.getProjectResource().getGoogleProjectId();
```

Comment:
It's probably unrelated, but this looks a bit similar to `GoogleProjectService.projectIdForFile()` which is used by an earlier step in this flight, `IngestFileGetProjectStep`.